store/store.go

```diff
@@ -59,15 +60,20 @@ func (s *DefaultStore) SaveBlock(block *types.Block, commit *types.Commit) error
 
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
-	// TODO(tzdybal): use transaction for consistency of DB (https://github.com/celestiaorg/optimint/issues/80)
-	err = multierr.Append(err, s.db.Set(getBlockKey(hash), blockBlob))
-	err = multierr.Append(err, s.db.Set(getCommitKey(hash), commitBlob))
-	err = multierr.Append(err, s.db.Set(getIndexKey(block.Header.Height), hash[:]))
+
+	bb := s.db.NewBatch()
+	err = multierr.Append(err, bb.Set(getBlockKey(hash), blockBlob))
+	err = multierr.Append(err, bb.Set(getCommitKey(hash), commitBlob))
+	err = multierr.Append(err, bb.Set(getIndexKey(block.Header.Height), hash[:]))
 
 	if err != nil {
 		return err
```

Comment:
As mentioned in previous commit, I think error should be simply propagated (without internal `Discard` call) from `Set` and here, we should call `bb.Discard()` before returning error.